### ISSUE TITLE
Fix arithmetic on usages

### ISSUE THREAD

**Issue by monoidal** [tweag/ghc#423]
In the branch, I made ⊥ * ω = ⊥. This is not sound:

```
{-# LANGUAGE GADTs, LinearTypes, ScopedTypeVariables, EmptyCase #-}
module N where

data Void
data U a where U :: a -> U a

elim :: U a ->. ()
elim (U _) = ()

f :: a ->. ()
f x = elim (U (\(a :: Void) -> case a of {}))
```

In the train, we came to the conclusion that ⊥ * ω should be a different usage. It must be a subusage of 0, but not of 1. It should not be 0, since then 0 = ⊥ * ω = ⊥ * (ω + 1) = ⊥ * ω + ⊥ = ⊥.

Additionally, ⊥ * 1 should be ⊥. Note that sup ⊥ 1 = 1, so sup and * are not the same on usages.

**Comment by monoidal** [comment]
The above is wrong: ⊥ * ω should be 0; distributivity does not hold in this form.

I've fixed the soundness bug in f4bc8196f. Remaining work:
(a) add a test
(b) put this on a consistent footing. Known options for now are (a) keep lists of `Zero | MUsage Multiplicity` in the usage env and do operations pairwise; (b) remove multiplicity polymorphism.

**Comment by monoidal** [comment]
Point (b) moved to upstream:#18768.
